#SOAL 1

In [8]:
import pandas as pd
import numpy as np

data = {
    'Tipe Ujian' : ['UTS1','UAS1']*3,
    'Mata Pelajaran' : ['Statistika','Fisika','Fisika','Statistika','Fisika','Statistika'],
    'Jumlah Peserta' : [14,17,12,16,14,13]
}
df = pd.DataFrame(data).set_index('Tipe Ujian')
df

Mata Pelajaran  Jumlah Peserta
Tipe Ujian                               
UTS1           Statistika              14
UAS1               Fisika              17
UTS1               Fisika              12
UAS1           Statistika              16
UTS1               Fisika              14
UAS1           Statistika              13

In [14]:
pd.pivot_table(df, values = 'Jumlah Peserta', index = 'Tipe Ujian', columns = 'Mata Pelajaran', aggfunc = 'sum')

Mata Pelajaran  Fisika  Statistika
Tipe Ujian                        
UAS1                17          29
UTS1                26          14

#Soal 2

In [62]:
import pandas as pd
data1 = {
    'Hari' : ['Sabtu','Minggu'],
    'Kebun Binatang' : [271,399],
    'Dufan' : [501,700],
    'Disney': [1000,1001],
    'Bali' : [900,803]
}
df1 = pd.DataFrame(data1).set_index('Hari')
df1

Hari  Kebun Binatang  Dufan  Disney  Bali
0   Sabtu             271    501    1000   900
1  Minggu             399    700    1001   803

In [67]:
pd.melt(df1.reset_index(), id_vars=['Hari'],value_vars=['Kebun Binatang','Dufan','Disney','Bali'],var_name='Tempat Pengunjung', value_name='Pengunjung',ignore_index = True)

Hari Tempat Pengunjung  Pengunjung
0   Sabtu    Kebun Binatang         271
1  Minggu    Kebun Binatang         399
2   Sabtu             Dufan         501
3  Minggu             Dufan         700
4   Sabtu            Disney        1000
5  Minggu            Disney        1001
6   Sabtu              Bali         900
7  Minggu              Bali         803

#Soal 3

In [69]:
df_d= pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Week 7/diamonds.csv')

In [74]:
df_d.groupby(['cut'])['price','carat'].agg({'price': lambda x:max(x)-min(x),'carat':'std'})

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  """Entry point for launching an IPython kernel.


price     carat
cut                       
Fair       18237  0.516404
Good       18461  0.454054
Ideal      18480  0.432876
Premium    18497  0.515262
Very Good  18482  0.459435

Berapa selisih harga tertinggi dan terendah dari diamond berkualitas Premium?

Ans: 18497